src/Nethermind/Nethermind.Consensus/Validators/UnclesValidator.cs

```diff
@@ -42,9 +42,10 @@ public bool Validate(BlockHeader header, BlockHeader? parentHeader, BlockHeader[
             for (int i = 0; i < uncles.Length; i++)
             {
                 BlockHeader uncle = uncles[i];
-                if (!_headerValidator.Validate(uncle, parentHeader, true, out _))
+                BlockHeader? uncleParent = _blockTree.FindParentHeader(uncle, BlockTreeLookupOptions.TotalDifficultyNotNeeded);
```

Comment:
Uncle parent should be the same as this block parent? Is it possible not to be the same? If yes shouldn't we check it? Maybe just checking hashes would suffice and we could skip this `FindParentHeader` call?